### DISCUSSION TITLE
Alternative for `tokio::net::TcpListener.incoming().poll_next(...)` in tokio 0.3

### DISCUSSION THREAD

**Discussion by Urhengulas** [tokio-rs/tokio#2983] [Q&A] [answered]
How can `tokio::net::TcpListener.incoming().poll_next(...)` from `tokio 0.2`, be replaced in `tokio 0.3`?

_Use case: https://github.com/google/tarpc/blob/d27f341bdebe29df927b6d2b6f1a26fd657a0efe/tarpc/src/serde_transport.rs#L273_

**Comment by taiki-e** [comment, marked as answer]
You can use [`poll_accept`](https://docs.rs/tokio/0.3.0/tokio/net/struct.TcpListener.html#method.poll_accept), like:

```rust
let next = ready!(Pin::new(&mut self.as_mut().project().listener).poll_accept(cx)?).0;
```

**Reply by Urhengulas** [reply]
> You can use [`poll_accept`](https://docs.rs/tokio/0.3.0/tokio/net/struct.TcpListener.html#method.poll_accept), like:
> 
> ```rust
> let next = ready!(Pin::new(&mut self.as_mut().project().listener).poll_accept(cx)?).0;
> ```

Just noticed that I meant `TcpListener` and not `TcpStream`, but I guess the answer works anyways...
